### ISSUE TITLE
JIT bug under differentiation w.r.t custom kernel parameters

### ISSUE THREAD

**Issue by michael-0brien** [geoffroyO/nufftax#13]
In `test_custom_kernel.py`, we have the test

```python
def test_grad_wrt_kernel_param_pure_jax_matches_fd(rng):
    M, nf = 40, 64
    x = jnp.asarray(rng.uniform(-np.pi, np.pi, M))
    c = jnp.asarray(rng.standard_normal(M)).astype(jnp.complex64)
    g = jnp.asarray(rng.standard_normal(nf))

    def loss(sigma):
        return jnp.sum(g * jnp.real(spread_1d_impl(x, c, nf, _gauss_kernel_from_sigma(sigma))))

    sigma = jnp.float32(1.5)
    grad = float(jax.grad(loss)(sigma))

    eps = 1e-3
    fd = (float(loss(sigma + eps)) - float(loss(sigma - eps))) / (2 * eps)
    assert_allclose(grad, fd, rtol=2e-2, atol=2e-2)
```

If `loss` is changed to 

```python
@jax.jit
def loss(sigma):
    # JIT compile and call `nufftax.core.spread_1d` instead of `nufftax.spread_1d_impl` function
    return jnp.sum(g * jnp.real(spread_1d(x, c, nf, _gauss_kernel_from_sigma(sigma))))
```

This returns

"""
Detected differentiation of a custom_vjp function with respect to a closed-over value. That isn't supported because the custom VJP rule only specifies how to differentiate the custom_vjp function with respect to explicit input parameters. Try passing the closed-over value into the custom_vjp function as an argument, and adapting the custom_vjp fwd and bwd rules.
"""

The error I was getting when applied to my full workflow, rather than this test, was

"""
Found a JAX Tracer object passed as an argument to a custom_vjp function in a position indicated by nondiff_argnums as non-differentiable. Tracers cannot be passed as non-differentiable arguments to custom_vjp functions; instead, nondiff_argnums should only be used for arguments that can't be or contain JAX tracers, e.g. function-valued arguments. In particular, array-valued arguments should typically not be indicated as nondiff_argnums.
"""

**Comment by geoffroyO** [comment]
Fixed by #15 (merged).
